capsules/src/usb/usbc_client.rs

```diff
@@ -56,22 +48,57 @@ pub struct Client<'a, C: 'a> {
 }
 
 impl<'a, C: hil::usb::UsbController<'a>> Client<'a, C> {
-    pub fn new(controller: &'a C) -> Self {
-        Client {
-            client_ctrl: ClientCtrl::new(
-                controller,
+    pub fn new(controller: &'a C, max_ctrl_packet_size: u8) -> Self {
+        let interfaces: &mut [descriptors::InterfaceDescriptor] =
+            &mut [descriptors::InterfaceDescriptor {
+                interface_number: 0,
+                alternate_setting: 0,
+                num_endpoints: 0,      // (excluding default control endpoint)
+                interface_class: 0xff, // vendor_specific
+                interface_subclass: 0xab,
+                interface_protocol: 0,
+                string_index: 0,
+            }];
+
+        let endpoints: &[&[EndpointDescriptor]] = &mut [&[
+            EndpointDescriptor {
+                endpoint_address: EndpointAddress::new_const(1, TransferDirection::DeviceToHost),
+                transfer_type: TransferType::Bulk,
+                max_packet_size: 8,
+                interval: 100,
+            },
+            EndpointDescriptor {
+                endpoint_address: EndpointAddress::new_const(2, TransferDirection::HostToDevice),
+                transfer_type: TransferType::Bulk,
+                max_packet_size: 8,
+                interval: 100,
```

Comment:
These aren't your fault, but we might as well fix them now so they stop perpetuating.

```suggestion
                interval: 0,
            },
            EndpointDescriptor {
                endpoint_address: EndpointAddress::new_const(2, TransferDirection::HostToDevice),
                transfer_type: TransferType::Bulk,
                max_packet_size: 8,
                interval: 0,
```